aws_xray_sdk/core/recorder.py

```diff
@@ -274,6 +274,20 @@ def current_segment(self):
             return entity.parent_segment
         else:
             return entity
+    
+    def _begin_subsegment_helper(self, name):
+        '''
+        Helper method to begin_subsegment and begin_subsegment_without_sampling
+        '''
+        result = True
```

Comment:
I feel this is still problematic. I'm not sure I get the point of using `result` and returning it.
Also, `result` seem to be of variable types. Here it is a boolean but later it can be a Subsegment. This can easily confusion and bugs.

I believe there is a better way to leverage this helper method than it's callers having to check its return types and values.
`_begin_subsegment_helper` can accept a parameter as flag for sampling or not. The callers can simply call this method with the flag and get back a Subsegment depending on the flag's value. How does that sound?
The method definition could look like:
```python
def _begin_subsegment_helper(name, namespace='local', beginWithoutSampling=False)
```